base_layer/wallet/src/ffi.rs

```diff
@@ -120,284 +435,574 @@ pub unsafe extern "C" fn create_pending_transaction_outputs(
         tx_id,
         outputs_to_be_spent: Vec::new(),
         outputs_to_be_received: Vec::new(),
-        timestamp: NaiveDateTime::parse_from_str("timestamp", "THE FORMAT WE CHOOSE").unwrap_or(NaiveDateTime::now), /* Use the rfc-3339 Format for this. */
+        timestamp: NaiveDateTime::parse_from_str("timestamp", "THE FORMAT WE CHOOSE").unwrap(), /* Use the rfc-3339 Format for this. */
     }))
 }
 
-/// Append an UnblindedOutput to be spent to the pending transaction outputs object
 #[no_mangle]
-pub unsafe extern "C" fn add_output_to_spend(
-    pending_tx: *mut PendingTransactionOutputs,
-    value: c_ulonglong,
-    spending_key: *const c_char,
-    feature_flags: c_char, // OutputFlags,
-    maturity: c_ulonglong, // u64
+pub unsafe extern "C" fn destroy_pending_transaction_outputs(pto: *mut PendingTransactionOutputs) {
+    if !pto.is_null() {
+        Box::from_raw(pto);
+    }
+}
+/// -------------------------------------------------------------------------------------------- ///
+
+/// -------------------------------- Compound Inputs, Outputs, Kernels ------------------------- ///
+/// Initialize a Transaction struct to be populated
+
+pub struct TransactionInputs(Vec<TransactionInput>);
+pub struct TransactionOutputs(Vec<TransactionOutput>);
+pub struct TransactionKernels(Vec<TransactionKernel>);
```

Comment:
I think this should just be a single Transaction object rather than split this into 3 separate pieces.

Reply:
You currently have 3 method to create new versions of each of these and 3 to destroy each where you could have just one `create_transaction` and one `destroy_transaction`